src/middlewared/middlewared/plugins/pool_/unlock.py

```diff
@@ -58,24 +60,31 @@ async def unlock_restarted_vms(self, dataset_name):
                 if not path:
                     continue
 
-                if path.startswith(zvol_name_to_path(dataset_name) + '/') or path.startswith(f'/mnt/{dataset_name}/'):
+                if (
```

Comment:
@sonicaj can you please ensure that this is indeed the correct logic? Can VM be consuming a FILESYSTEM?

Reply:
@Qubad786 too since he has done some work in this area

Reply:
VM can yes consume a FILESYSTEM with `RAW` devices so this LGTM